### PR TITLE
Added overview of C# equality operators

### LINKED ISSUE #10732: Abridged Reference: operators / equality operators
The equality operators are:

- `==` (equal)
- `!=` (not equal)

Sub-tasks are:

- [ ] Create the new articles
- [ ] Search for existing issues on content being migrated and consolidated. This is a good opportunity to fix those.
- [ ] Redirect existing content to the new articles.
- [ ] Remove old articles.
- [ ] Update all links in the .NET Content

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #10732
Existing code snippets are reused, so no updates in dotnet/samples.

### CODE REVIEW

File: docs/csharp/language-reference/operators/equality-operators.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+The equality operator `==` returns `true` if its operands are equal, `false` otherwise.
+
+### Value types equality
+
+Operands of the [built-in value types](../keywords/value-types-table.md) are equal if their values are equal:
+
+[!code-csharp-interactive[value types equality](~/samples/snippets/csharp/language-reference/operators/EqualityAndNonEqualityExamples.cs#ValueTypesEquality)]
+
+> [!NOTE]
+> For equality and relational operators `==`, `>`, `<`, `>=`, and `<=`, if any of the operands is not a number (<xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType>), the result of operation is `false`. That means that the `NaN` value is neither greater than, less than, nor equal to any other `double` (or `float`) value. For more information and examples, see the <xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType> reference article.
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/equality-operators.md]: Not related to you PR. We should add to this note that `double.NaN` is not equal to `double.NaN`:

Comment (BillWagner) [inline docs/csharp/language-reference/operators/equality-operators.md]:
```suggestion
> For equality and relational operators `==`, `>`, `<`, `>=`, and `<=`, if any of the operands is not a number (<xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType>), the result of operation is `false`. That means that the `NaN` value is neither greater than, less than, nor equal to any other `double` (or `float`) value, including `NaN`. For more information and examples, see the <xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType> reference article.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
I'm always impressed by the thoroughness of the work you do in our repository @pkulikov . Thank you.

I had one comment, that's not related to this PR.  Let me know what you think, and then we'll :shipit:

File: docs/csharp/language-reference/operators/equality-operators.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/equality-operators.md]: Thanks! Hesitated about the same, when reviewed it.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @pkulikov 
:shipit: now!

**Comment by pkulikov** [comment]
@BillWagner thanks!
Logical operators are next to come.

**Pull request merged**